src/core/mkMergedShell.nix

```diff
@@ -0,0 +1,57 @@
+{ lib, pkgs, ... }:
+
+let
+  utils = lib.iogx.utils;
+
+
+  mergeTwoShellProfiles = p1: p2: {
+
+    packages =
+      utils.getAttrWithDefault "packages" [ ] p1 ++
+      utils.getAttrWithDefault "packages" [ ] p2;
+
+    scripts =
+      let
+        scripts1 = utils.getAttrWithDefault "scripts" { } p1;
+        scripts2 = utils.getAttrWithDefault "scripts" { } p2;
+      in
+      # TODO check clashes
+      scripts1 // scripts2;
```

Comment:
@coderabbitai The TODO comment already indicates that the author is aware of the need for further work here. It might be valuable to flag the TODO comment for other reviewers, but it's redundant to rearticulate the future work identified in the comment.